Create docs connection configuration (#137)
astronomer/astronomer-cosmos: docs/dbt/connections.rst

Connections
===========

Cosmos supports these common dbt core adapters. However, unlike running dbt core from the CLI, Cosmos does *not* need a
profiles.yml file to authenticate. Instead, you will create an Airflow connection object for your Database and pass it
to the desired Cosmos component.

.. tabs::

   .. tab:: Snowflake

        .. note::
            Cosmos currently only supports User/Password authentication to Snowflake. Additional authentication methods will be
            added soon. If there is an authentication method that your organization requires, open a PR/Issue on the Cosmos GitHub
            repo `here <https://github.com/astronomer/astronomer-cosmos>`_.

>         Here are the specific steps that you should take if you are configuring Cosmos to run against Snowflake:

Reviewer: ```suggestion
        Follow these steps when configuring Cosmos to run against Snowflake:
```

            .. code-block:: text

                AIRFLOW_CONN_SNOWFLAKE_DEFAULT=snowflake://user123:p@ssword!@/my_db_schema?__extra__={"account":+"gp54783",+"warehouse":+"my_warehouse",+"database":+"my_db",+"region":+"us-east-1",+"role":+"user123",+"insecure_mode":+false}

        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
        Class via the `conn_id` parameter:

            .. code-block:: python

                jaffle_shop = DbtTaskGroup(
                    ...
                    conn_id="snowflake_default",
                )

        .. note::
            If you left the `Schema` field blank when configuring your Snowflake connection or if you would like to override it,
            then specify the schema parameter via the `dbt_args`:

            .. code-block:: python

                jaffle_shop = DbtTaskGroup(
                    conn_id="snowflake_default",
                    dbt_args={
                        "schema": "my_schema",
                    }
                )

   .. tab:: Redshift

        .. note::

>             Cosmos currently only supports User/Password authentication to Redshift. Additional authentication methods will be

Reviewer: @pohek321 how hard would it be to add other auth systems to cosmos? Do we have issues created for this?

Author: @dimberman we'll have to identify the Airflow connection fields to determine if the user is doing IAM vs. user/password, then we'll have to map the Airflow connection fields to the appropriate profiles.yml fields. Opened an issue for different auth methods already:
- [Add authentication method for Redshift (IAM Authentication)](https://github.com/astronomer/astronomer-cosmos/issues/44)
- [Add authentication method for Snowflake (Key Pair Authentication)](https://github.com/astronomer/astronomer-cosmos/issues/42)
- [Add authentication method for BigQuery (Service Account File)](https://github.com/astronomer/astronomer-cosmos/issues/43)

Author: Marking this as resolved since we have issues tracking various auth methods that we plan on adding.

            .. code-block:: text

                AIRFLOW_CONN_REDSHIFT_DEFAULT=redshift://username:password@redshift-cluster.endpoint.us-west-2.redshift.amazonaws.com:5439/db_name

        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
        Class via the `conn_id` parameter:

        .. note::
            Since the Airflow Connections for redshift do not provide a field for a default schema parameter, you will need to
            add it via the `dbt_args`:

        .. code-block:: python

            jaffle_shop = DbtTaskGroup(
                conn_id="redshift_default",
                dbt_args={
                    "schema": "my_schema",
                }
            )

   .. tab:: BigQuery

        .. note::
            Cosmos currently only supports Service Account JSON method when authenticating to BigQuery. Additional
            authentication methods will be added soon. If there is an authentication method that your organization requires, open a
            PR/Issue on the Cosmos GitHub repo `here <https://github.com/astronomer/astronomer-cosmos>`_.

>         Here are the specific steps that you should take if you are configuring Cosmos to run against BigQuery:

Reviewer: ```suggestion
        Follow these steps when configuring Cosmos to run against BigQuery:
```

        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
        Class via the `conn_id` parameter:

        .. note::
            Since the Airflow Connections for BigQuery do not provide a field for a default dataset parameter, you will need to
            add it via the `dbt_args`:

        .. code-block:: python

            jaffle_shop = DbtTaskGroup(
                conn_id="bigquery_default",
                dbt_args={
                    "schema": "your_bigquery_dataset",
                }
            )

   .. tab:: Databricks

>         Here are the specific steps that you should take if you are configuring Cosmos to run against Databricks:

Reviewer: ```suggestion
        Follow these steps when configuring Cosmos to run against Databricks:
```

        for this example would look like this:

            .. note::
                The following variable value would need to be encoded. You can use a service like `urlencoder.org <https://www.urlencoder.org/>`_ to do that.

            .. code-block:: text

                AIRFLOW_CONN_DATABRICKS_DEFAULT=databricks://adb-1234567891011.12.azuredatabricks.net/your_databricks_catalog?http_path=/sql/protocolv1/o/1234567891011/0503-58462-kdw76lbv&token=<your-databricks-token>

        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
        Class via the `conn_id` parameter:

        .. note::
            Since the Airflow Connections for databricks do not provide a field for a default schema parameter, you will need to
            add it via the `dbt_args`:

        .. code-block:: python

            jaffle_shop = DbtTaskGroup(
                conn_id="databricks_default",
                dbt_args={
                    "schema": "your_databricks_db",
                }
            )

   .. tab:: Postgres

>         Here are the specific steps that you should take if you are configuring Cosmos to run against Postgres:

Reviewer: ```suggestion
        Follow these steps when configuring Cosmos to run against Postgres:
```

               :width: 800

               Creating a postgres connection in Airflow

        (Optional) If your team prefers storing connections as environment variables or in a supported secrets backend, the connection URI
        for this example would look like this:

            .. code-block:: text

                AIRFLOW_CONN_POSTGRES_DEFAULT=postgres://your_postgres_username:password@34.29.167.133:5432/your_postgres_db_name

        3. With the dependencies and connection in place, you can pass this connection to a DbtTaskGroup, DbtDag, or Dbt Operator
        Class via the `conn_id` parameter:

        .. note::
            Since the Airflow Connections for postgres do not provide a field for a default schema parameter, you will need to
            add it via the `dbt_args`:

        .. code-block:: python

            jaffle_shop = DbtTaskGroup(
                conn_id="postgres_default",
                dbt_args={
                    "schema": "my_schema",
                }
            )

Overriding connection Parameters
--------------------------------

> Although setting up a database name in your Airflow Connection is required, The database/schema values from the

Reviewer: ```suggestion
To use Airflow with Cosmos, you must set up a database name in your Airflow Connection. However, when you instantiate a Cosmos Object, you can choose to override the database/schema values from the Connection object.```
